source/tests/unit/ni_fake_service_tests.cpp

```diff
@@ -1822,6 +1822,64 @@ TEST(NiFakeServiceTests, NiFakeService_AcceptViSessionArray_CallsAcceptViSession
   EXPECT_EQ(kDriverSuccess, response.status());
 }
 
+// Test for two-dimension mechanism
```

Comment:
nit: I think this is a helpful place to emphasize that `two-dimension` ONLY does the size validation check and passes through the data. (Unlike other size mechanisms that do stuff to figure out the size).

Reply:
I had forgotten to go back and update this but have now added a more full description.